test/TestCases/common/index.js

```diff
@@ -118,18 +118,21 @@ exports.dotnetModule = process.env.TEST_DOTNET_MODULE_PATH;
 exports.dotnetVersion = process.env.TEST_DOTNET_VERSION;
 
 exports.loadDotnetModule = function () {
-  // The Node API module may need the require() function at initialization time; passing it as
-  // a global is the best (only?) solution, since it cannot be obtained via any `napi_*` function.
-  global.require = require;
-
   let dotnetModule;
   if (exports.dotnetHost) {
-    // Normally the index.js script in the npm package takes care of locating the correct
+    // Normally the init.js script in the npm package takes care of locating the correct
     // native host and managed host binaries for the current environment.
     dotnetModule = require(exports.dotnetHost)
-      .initialize(exports.dotnetVersion, exports.dotnetHost.replace(/\.node$/, '.DotNetHost.dll'))
+      .initialize(
+        exports.dotnetVersion,
+        exports.dotnetHost.replace(/\.node$/, '.DotNetHost.dll'),
+        require)
       .require(exports.dotnetModule);
   } else {
+    // The Node API module may need the require() function at initialization time; passing it as a
+    // global is the only solution for AOT, since it cannot be obtained via any `napi_*` function.
+    global.require = require;
```

Comment:
>require [](http://example.com/codeflow?start=11&length=7)

Can we use a different name on the global that pseudo-scopes it to our project?
E.g. __node_api_dotnet_require.

Reply:
I'm not sure that would be better. This actually has to be set by user code at the moment, so it might be a little strange if user code had to set a property with that scoped name.

Later we might want to have the TS generator also output a `.js` wrapper for the AOT `.node` module. That would take care of both referencing the `.d.ts` and setting this global.

Reply:
The problem with the global name `require` is that it will affect all modules in a Node.JS applications.
In some cases, in may cause some unexpected behaviors that will be in the end point as the root cause to our code and security experts will mark our NPM package as unsecure.
It must be any other name that must not cause side effects in other modules.
Any global name is callable from any module as it was locally defined in that module.
Having a special name that is visually related to our module is very important in such scenario.

Reply:
BTW, we had discussed the issue of passing `module` parameter to the new entry point and generally it was not rejected. It was considered that it could be part of the new Node-API version. Though, it may take a year or more before we could use it.

Reply:
What if we create a special node-api-dotnet global property object and the `require` will be just its property?
E.g. `global.node_api_dotnet = { require };`

Reply:
Another alternative is to make sure that we always delete this property inside of the entry point function.
This way, I agree with you, it doesn't matter how we call it.
My main concern is that we must not leak this global property to anyone else.

Reply:
Updated to use `global.node_api_dotnet = { require }`.

> Another alternative is to make sure that we always delete this property inside of the entry point function.

I wouldn't want to _delete_ `global.require`, as some other code may rely on it. It is somewhat likely for applications or frameworks to set `global.require = require`, because Node.js itself does that in some contexts. For example:
```
$ node -p "typeof global.require"
function
```